posts/tests.py

```diff
@@ -1,3 +1,88 @@
-# from django.test import TestCase
+import pytest
+from .models import Poll, Post
+from django.contrib.auth.models import User
 
-# Create your tests here.
+USERNAME = "testuser"
+FIRSTNAME = "Test"
+LASTNAME = "User"
+PASSWORD = "testpass"
+EMAIL = "testuser@gmail.com"
+DESCRIPTION = "this is a test poll"
+FILE1 = "Alon_Shakaroffs_resume.pdf"
+FILE2 = "Olive.png"
+CHOICETEXT1 = "First option"
+CHOICETEXT2 = "Second option"
+
+
+@pytest.fixture
+def new_user():
+    return User(username=USERNAME, first_name=FIRSTNAME, last_name=LASTNAME, password=PASSWORD, email=EMAIL)
+
+
+@pytest.fixture
+def persist_user(new_user):
+    new_user.save()
+    return new_user
+
+
+@pytest.fixture
+def new_poll(new_user):
+    return Poll(author=new_user, description=DESCRIPTION)
+
+
+@pytest.fixture
+def persist_Poll(new_poll):
+    new_poll.author.save()
+    new_poll.save()
+    return new_poll
+
+
+# -------------------------------------------- Poll tests --------------------------------------------
+
+
+@pytest.mark.django_db()
+class TestPoll:
+    # Check that the Poll values are the same as the Poll inputs.
+    def test_new_poll(self, new_poll):
+        assert new_poll.author.username == USERNAME
+        assert new_poll.description == DESCRIPTION
+
+    # Check if the Poll is saved in the database and accessible via its author.
+    def test_persist_poll(self, persist_Poll):
+        assert persist_Poll in Poll.objects.all()
+
+    # Check if the Poll can be referred as post and accessible from Post list .
+    def test_poll_is_post(self, persist_Poll):
+        assert isinstance(persist_Poll, Post)
+        assert len(Post.objects.all()) == 1
+        post = Post.objects.all().first()
+        assert hasattr(post, 'poll') is True
+        assert hasattr(post, 'resume') is False
+        poll = post.poll
+        assert persist_Poll == poll
+
+    # Check if Poll deletion delete only Poll from database.
+    def test_delete_poll(self, persist_Poll):
+        assert persist_Poll in Poll.objects.all()
+        assert persist_Poll.author in User.objects.all()
+        persist_Poll.delete()
+        assert persist_Poll not in Poll.objects.all()
+        assert persist_Poll.author in User.objects.all()
+
+    # Check if Poll's author deletion delete both the author and the poll.
+    def test_delete_polls_aouthor(self, persist_Poll):
```

Comment:
There's a type in the method name, its suppose to be author.

Reply:
Fixed